bndtools.core/src/bndtools/central/sync/WorkspaceChange.java

```diff
@@ -0,0 +1,63 @@
+package bndtools.central.sync;
+
+import java.util.Collection;
+
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.resources.IncrementalProjectBuilder;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import aQute.bnd.build.Workspace;
+import aQute.bnd.service.RepositoryPlugin;
+
+@Component
+public class WorkspaceChange {
+	final static IWorkspace	workspace	= ResourcesPlugin.getWorkspace();
+
+	@Reference
+	Workspace				ws;
+
+	boolean					first		= true;
+	volatile boolean		cancel		= false;
+
+	@Activate
+	void activate() {
+		ws.on("workspace-changes")
+			.initial(this::init)
+			.repositoriesReady(this::done);
```

Comment:
I am not sure I see the distinction here. init and repos are only called together (in Workspace.forceInitialization). So init is called, sets cancel to true, repos is called and sets cancel to false before scheduling the job. I am not sure how done0 ever sees cancel==true?

Reply:
It handles the case when there is init -> repo-ready -> start job after 2 s -> init! -> enter job

It aborts a job if we're bound to receive another one. Maybe not such a likely case but it can happen.